From 46a65cdd5beaaca05ccbe65279e667727306e429 Mon Sep 17 00:00:00 2001
From: huyan <hu.huyan@huawei.com>
Date: Mon, 17 Jun 2019 11:57:26 +0800
Subject: [PATCH] backport remove resource leak from nfs/mydaemon.c

---
 support/nfs/mydaemon.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/support/nfs/mydaemon.c b/support/nfs/mydaemon.c
index 343e80b..d1cf08d 100644
--- a/support/nfs/mydaemon.c
+++ b/support/nfs/mydaemon.c
@@ -123,6 +123,7 @@ daemon_init(bool fg)
 	dup2(tempfd, 0);
 	dup2(tempfd, 1);
 	dup2(tempfd, 2);
+	close(tempfd);
 	closelog();
 	dup2(pipefds[1], 3);
 	pipefds[1] = 3;
-- 
1.8.3.1

