From 4f9d2bf8a677a61a105703319e9f7985d20f6f02 Mon Sep 17 00:00:00 2001
From: huyan <hu.huyan@huawei.com>
Date: Mon, 17 Jun 2019 12:10:41 +0800
Subject: [PATCH] backport remove resource leak from nfs/svc_socket.c

---
 support/nfs/svc_socket.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/support/nfs/svc_socket.c b/support/nfs/svc_socket.c
index d56507a..c711b39 100644
--- a/support/nfs/svc_socket.c
+++ b/support/nfs/svc_socket.c
@@ -134,6 +134,7 @@ svc_socket (u_long number, int type, int protocol, int reuse)
       if (ret < 0)
 	{
 	  xlog(L_ERROR, "svc_socket: socket reuse problem: %m");
+	  (void) __close(sock);
 	  return ret;
 	}
     }
-- 
1.8.3.1

